src/NLog/Internal/FileAppenders/UnleashedMultiProcessFileAppender.cs

```diff
@@ -0,0 +1,215 @@
+
+                _file = null;
+
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Writes the specified bytes.
+        /// </summary>
+        /// <param name="bytes">The bytes to be written.</param>
+        public override void Write(byte[] bytes)
+        {
+            try
+            {
+                uint written;
+                bool success = Win32FileNativeMethods.WriteFile(_file.DangerousGetHandle(), bytes, (uint)bytes.Length, out written, IntPtr.Zero);
+                if (!success || (uint)bytes.Length != written)
+                {
+                    Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
+                }
+                FileTouched();
+            }
+            finally
+            {
+            }
+        }
+
+        /// <summary>
+        /// Closes this instance.
+        /// </summary>
+        public override void Close()
+        {
+            InternalLogger.Trace("Closing '{0}'", FileName);
+            if (_file != null && !_file.IsClosed)
+            {
+                _file.Close();
+            }
+
+            _file = null;
+            FileTouched();
+        }
+
+        /// <summary>
+        /// Flushes this instance.
+        /// </summary>
+        public override void Flush()
+        {
+            // do nothing, the file is written directly
+        }
+
+        /// <summary>
+        /// Gets the file info.
+        /// </summary>
+        /// <returns>The file characteristics, if the file information was retrieved successfully, otherwise null.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Runtime.InteropServices.SafeHandle.DangerousGetHandle", Justification = "Optimization")]
+        public override FileCharacteristics GetFileCharacteristics()
+        {
+            return FileCharacteristicsHelper.Helper.GetFileCharacteristics(FileName, _file.DangerousGetHandle());
+        }
+
+        /// <summary>
+        /// Factory class.
+        /// </summary>
+        private class Factory : IFileAppenderFactory
+        {
+            /// <summary>
+            /// Opens the appender for given file name and parameters.
+            /// </summary>
+            /// <param name="fileName">Name of the file.</param>
+            /// <param name="parameters">Creation parameters.</param>
+            /// <returns>
+            /// Instance of <see cref="BaseFileAppender"/> which can be used to write to the file.
+            /// </returns>
+            BaseFileAppender IFileAppenderFactory.Open(string fileName, ICreateFileParameters parameters)
+            {
+                try
+                {
+                    if (!parameters.ForceManaged && PlatformDetector.IsDesktopWin32)
+                        return new UnleashedMultiProcessFileAppender(fileName, parameters);
+                }
+                catch
```

Comment:
catch is just so that the fallback to MutexMultiProcessFileAppender will get its chance. So I'll place the return "new MutexMultiProcessFileAppender(..)" directly into the catch block preceded with an internal debug log.